### PR TITLE
fix(docs): minor UX Writing Guide copy to match Dev Center IA conventions

### PR DESCRIPTION
**Pull request by slsriehl**
## What?

Describe what changes you are making. e.g. `Adds X prop in the Box component.`
* upcase titles

## Why?

Please provide the reason(s) why you are making this change? Please no one-liners like, `It was missing.`; expand upon the reason like, `Adds X prop because it was missing in the Box component. Documentation notes this prop exists, but it actually wasn't implemented.`
* the Dev Center uses title case titles and sentence case headings inside each doc. the Guide now conforms to that.

## Screenshots/Screen Recordings

This is a component library so we love visually looking at changes! If this applies to your pull request, show us your hard work in action.

## Testing/Proof

Explain how you tested your new feature or bugfix. Please refrain from one-liners like, `Added tests.`; expand upon how you tested your changes like, `Added local tests to the new component and ensured test coverage was met. Manually tested on Safari@x.x.x.`.

ping @bc-ghough @davelinke

### CODE REVIEW

**Review by chanceaclark — APPROVED** [review]

**Comment by slsriehl** [comment]
converted to draft and do not merge label added -- merge or close pending review and approval from @bigcommerce/product-design. cc @bc-ghough.

**Comment by bc-ghough** [comment]
Product Design has had a team sync (1/24) on this PR and want to reiterate that this guide is for apps, integrations and our control panel. It should not reflect the writing rules for the dev documentation. We also discussed the term "Content guide" and decided the section name should be reverted back to "UX Writing Guide as 'UX Writing' is very different than 'Content Guide'

File: packages/docs/design-docs/ux-writing/index.mdx
Comment (slsriehl) [inline packages/docs/design-docs/ux-writing/index.mdx:5]:
```suggestion
# UX Writing Guide
```

File: packages/docs/design-docs/ux-writing/index.mdx
Comment (slsriehl) [inline packages/docs/design-docs/ux-writing/index.mdx:26]:
```suggestion
The purpose of the **BigCommerce UX Writing Guide** is to present a consistent, coherent and competent voice to BigCommerce merchants and their shoppers as it relates to in-product, customer-facing communication, aka microcopy.
```

File: packages/docs/design-docs/ux-writing/the-basics.mdx
Comment (slsriehl) [inline packages/docs/design-docs/ux-writing/the-basics.mdx:73]:
```suggestion
The goal of this section of the UX Writing Guide is to give you a quick gloss on how to remove non-inclusive terms and phrases from the discourse of ecommerce.
```

**Comment by slsriehl** [comment]
Do not merge without approval from @bc-ghough

**Review by bc-ghough — APPROVED** [review]

**Review by markcmurphy — APPROVED** [review]

**Pull request merged**
